projects/cds/src/profiletag/services/profile-tag-push-events.service.ts

```diff
@@ -0,0 +1,150 @@
+import { Injectable } from '@angular/core';
+import {
+  ActiveCartService,
+  EventService,
+  PersonalizationContextService,
+} from '@spartacus/core';
+import {
+  CartPageVisited,
+  CategoryPageVisited,
+  HomePageVisited,
+  KeywordSearchPageVisited,
+  OrderConfirmationPageVisited,
+  PageVisited,
+  ProductDetailsPageVisited,
+} from 'projects/core/src/routing/event/routing.events';
+import { merge, Observable, of } from 'rxjs';
+import { map, skipWhile, withLatestFrom } from 'rxjs/operators';
+import {
+  CartChangedPushEvent,
+  CartViewPushEvent,
+  CategoryViewPushEvent,
+  HomePageViewPushEvent,
+  KeywordSearchPushEvent,
+  OrderConfirmationPushEvent,
+  PageViewPushEvent,
+  ProductViewPushEvent,
+  ProfileTagPushEvent,
+} from '../model/profile-tag.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProfileTagPushEventsService {
+  protected pushEvents$: Observable<ProfileTagPushEvent> = merge(
+    this.categoryPageVisited(),
+    this.productDetailsPageView(),
+    this.searchResultsChanged(),
+    this.homePageVisitedEvent(),
+    this.cartPageVisitedEvent(),
+    this.pageVisitedEvent(),
+    this.orderConfirmationVisited(),
+    this.cartChanged()
+  );
+  constructor(
+    protected activeCartService: ActiveCartService,
+    protected eventService: EventService,
+    protected personalizationContextService: PersonalizationContextService
+  ) {}
+
+  getPushEvents(): Observable<ProfileTagPushEvent> {
```

Comment:
As you already have the `pushEvents` property marked as `protected`, we should either move this code to it, or mark it as `private`.
Just to avoid potential confusion about what to override when extending the class.

Reply:
Set it to private